### PR TITLE
Auth0 all the things

### PR DESCRIPTION
**Pull request by mcsearchin**
This PR officially switches discovery UI to use auth0 for all authentication.

There will be a future card to make it 'configurable' - allowing us to still use default (Discovery API LDAP) auth in addition to Auth0.

As part of this effort we created a new higher-order component for exposing auth details to react components (`auth0-wrapper.js`)

### CODE REVIEW

File: src/auth/auth0-wrapper.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+const withAuth0 = WrappedComponent => {
+  const Auth0Wrapper = props => {
+    const [isAuthenticated, setAuthenticated] = useState()
+    const [isLoading, setLoading] = useState()
+    const [auth0Client, setAuth0Client] = useState()
+
+    useEffect(() => {
+      const connectAuth0 = async () => {
+        setLoading(true)
```
Comment (NathanielScottStevens) [inline src/auth/auth0-wrapper.js]: I would've assumed that it wouldn't be possible to read a property from this component before the empty useEffect had returned. It looks like that is not the case though and the useEffect is called after the wrapped component gets the props passed down. Can you help me understand that?

File: src/auth/auth0-wrapper.test.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+jest.mock('@auth0/auth0-spa-js')
+jest.mock('axios')
+
+const Wrapped = () => <div />
+
+const originalError = console.error
+beforeAll(() => {
+  console.error = (...args) => {
+    if (/Warning.*not wrapped in act/.test(args[0])) {
```
Comment (NathanielScottStevens) [inline src/auth/auth0-wrapper.test.js]: What is this about?

File: src/components/auth0-login-zone/auth0-login-zone.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      {
+        isAuthenticated
+          ? <button onClick={() => { logout({ returnTo }) }}><LoginSvgsAndText text="LOG OUT" /></button>
+          : <button onClick={loginWithRedirect}><LoginSvgsAndText text="LOG IN" /></button>
+      }
+    </login-zone>
+  )
+}
+
+Auth0LoginZone.propTypes = {
```
Comment (NathanielScottStevens) [inline src/components/auth0-login-zone/auth0-login-zone.js]: Thank you for giving it propTypes

File: src/pages/oauth-view/oauth-view.js [hunk 4]
```diff
@@ -1,46 +1,53 @@
...
-
+import { useEffect, useState } from 'react'
+import { Redirect } from 'react-router'
+import './oauth-view.scss'
+import routes from '../../routes'
+import qs from 'qs'
+import LoadingElement from '../../components/generic-elements/loading-element'
+import PropTypes from 'prop-types'
+
+const hasCodeParameter = search => {
```
Comment (NathanielScottStevens) [inline src/pages/oauth-view/oauth-view.js]: It's unclear to me what the "code" is

File: src/auth/auth0-wrapper.test.js (hunk 2 above)
Comment (mcsearchin) [reply src/auth/auth0-wrapper.test.js]: This is a crummy workaround I found for testing `useEffect` with enzyme's `mount` method.  `useEffect` works the way you would expect if you use `mount`, but it logs a nasty warning.  This setup just swallows that error.  It's currently also in the `oauth-view.test.js`.  I think we can probably remove it once we upgrade enzyme.  I thought about trying to move it up into a more global setup, but then I forgot.  😁

File: src/components/auth0-login-zone/auth0-login-zone.js (hunk 3 above)
Comment (mcsearchin) [reply src/components/auth0-login-zone/auth0-login-zone.js]: 👍

File: src/pages/oauth-view/oauth-view.js (hunk 4 above)
Comment (mcsearchin) [reply src/pages/oauth-view/oauth-view.js]: This is an authorization code.  We will rename it to reflect this.

File: src/auth/auth0-wrapper.js (hunk 1 above)
Comment (mcsearchin) [reply src/auth/auth0-wrapper.js]: I'm not sure I understand your question.  Let's chat about this in the morning.

File: src/search-params/search-params-manager.js [hunk 5]
```diff
@@ -96,8 +96,7 @@ SearchParamsManager.propTypes = {
   toggleApiAccessible: PropTypes.func.isRequired,
   updateSortOrder: PropTypes.func.isRequired,
   updateSearchText: PropTypes.func.isRequired,
-  updatePage: PropTypes.func.isRequired,
-  toggleFactes: PropTypes.func.isRequired
+  updatePage: PropTypes.func.isRequired
```
Comment (bbrewer-papa-x) [inline src/search-params/search-params-manager.js]: Oh, oops. We misspelled that pretty badly

**Pull request merged**